### PR TITLE
build: remove unnecessary codecov dependency & upgrade GHA

### PR DESCRIPTION
**Pull request by matthugs**
Context: https://about.codecov.io/blog/message-regarding-the-pypi-package/

### CODE REVIEW

File: requirements/ci.in [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Requirements for running tests in Travis
 -c constraints.txt
 
-codecov                   # Code coverage reporting
+coverage                  # Calculate code coverage of tests.
```
Comment (timmc-edx) [inline requirements/ci.in:4]: `ci.in` should only pull in tox and tox-battery -- tox manages its own virtualenv where it pulls in `coverage` (via `test.txt` I believe).

Comment (timmc-edx) [reply requirements/ci.in:4]: (so this line should just be deleted)

**Comment by timmc-edx** [comment]
Actually, it looks like Feanil beat you to it: https://github.com/openedx/django-user-tasks/pull/274

**Comment by matthugs** [comment]
Indeed @feanil did! Thanks @timmc-edx & Feanil!

**Pull request closed without merging**